MvvmCross-Forms/MvvmCross.Forms/Views/MvxFormsPagePresenter.cs

```diff
@@ -151,15 +152,34 @@ public override void RegisterAttributeTypes()
 
         public override void ChangePresentation(MvxPresentationHint hint)
         {
-            if (HandlePresentationChange(hint)) return;
-
-            if (hint is MvxClosePresentationHint)
+            if (hint is MvxPopToRootPresentationHint popToRootHint)
```

Comment:
`GetHostPageOfType<NavigationPage>()` is used in all the `if` statements. Can you just call this once and reuse?